tests/test_hamiltonians_trotter.py

```diff
@@ -143,6 +143,14 @@ def test_trotter_hamiltonian_three_qubit_term(backend):
     backend.assert_allclose(final_state, target_state)
 
 
+def test_symbolic_hamiltonian_circuit_different_dts(backend):
+    """Issue: https://github.com/qiboteam/qibo/issues/1357."""
+    ham = hamiltonians.SymbolicHamiltonian(symbols.Z(0))
+    a = ham.circuit(0.1)
+    b = ham.circuit(0.1)
+    assert np.linalg.norm(ham.circuit(0.2).unitary() - (a + b).unitary()) == 0
```

Comment:
```suggestion
    assert np.allclose(ham.circuit(0.2).unitary(), (a + b).unitary())
```
?

Reply:
suggestion looks good to me too

Reply:
784db2c0e3c5c6aad5de18be5574aa575814960c